packages/core/src/new-api/internal/module-builder.ts

```diff
@@ -294,6 +297,34 @@ export class IgnitionModuleBuilderImplementation<
     return future;
   }
 
+  public contractAt(
+    contractName: string,
+    address: string,
```

Comment:
I think this should be `string` or some future that represents a string. Otherwise I'd agree that this doesn't make sense.

There should be two ways of getting a future like that:
1. `staticCall`
2. Reading an event.

(2) is not well defined yet, but we should think about it asap.